### PR TITLE
Adding communication with other device and detect its address for commun...

### PR DESCRIPTION
**Pull request by Meli-0xFF**
Done

### CODE REVIEW

File: ball_detect/main.c [hunk 1]
```diff
@@ -32,6 +36,12 @@ void setup(void) {
 	
 	//LED REGISTER SETUP
 	DDRD |= 1 << 3;	
+
+	//DIP-SWITCH INITIALIZATION
+    DDRC = 0;
```
Comment (pepol) [inline ball_detect/main.c]: There are 4 spaces here, but a hard tab (shown as 8 spaces) elsewhere.

File: ball_detect/main.c [hunk 2]
```diff
@@ -32,6 +36,12 @@ void setup(void) {
...
+	PORTC |= \
```
Comment (pepol) [inline ball_detect/main.c]: Are those backslashes (`\`) necessary?

File: ball_detect/main.c [hunk 3]
```diff
@@ -54,10 +75,23 @@ void sendc(uint8_t ch) {
 }
 
 ISR(USART_RX_vect) {	
+	PORTD ^= 1 << 3;
+	uint32_t modulo, position;
+	uint8_t read_char = UDR0, i, j;
+	if(read_char == my_address) {
+		for(j = 2; j <= 0; j--) {
```
Comment (pepol) [inline ball_detect/main.c]: This loop won't run a single time (didn't you mean: `j >= 0`?).

File: ball_detect/main.c [hunk 4]
```diff
@@ -54,10 +75,23 @@ void sendc(uint8_t ch) {
...
 
 ISR(USART_RX_vect) {	
+	PORTD ^= 1 << 3;
+	uint32_t modulo, position;
+	uint8_t read_char = UDR0, i, j;
+	if(read_char == my_address) {
+		for(j = 2; j <= 0; j--) {	
+			modulo = vision_result[j];
+			position = 1000000;
+			for(i = 0;i < 7;i++) {
```
Comment (pepol) [inline ball_detect/main.c]: Spaces after semicolons, please.

**Comment by pepol** [comment]
@Meli-0xFF thanks for this. Please address my comments, otherwise looks good!

File: ball_detect/main.c [hunk 5]
```diff
@@ -32,6 +36,10 @@ void setup(void) {
 	
 	//LED REGISTER SETUP
 	DDRD |= 1 << 3;	
+
+	//DIP-SWITCH INITIALIZATION
+	DDRC = 0;
+	PORTC |= ((1 << DIP_SWITCH_1) | (1 << DIP_SWITCH_2) | (1 << DIP_SWITCH_3) | (1 << DIP_SWITCH_4) | (1 << DIP_SWITCH_5) | (1 << DIP_SWITCH_6));
```
Comment (pepol) [inline ball_detect/main.c]: Please wrap that line to be at most 80 characters long though.

**Comment by Meli-0xFF** [comment]
Done

**Comment by Meli-0xFF** [comment]
done

**Comment by Meli-0xFF** [comment]
done

**Comment by Meli-0xFF** [comment]
finally done

**Comment by pepol** [comment]
LGTM, please squash with `devices: Added detection and communication.` as summary line in commit message.

**Comment by pepol** [comment]
Thanks for this @Meli-0xFF!

**Pull request merged**